src/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddProjectsToSolutionPostAction.cs

```diff
@@ -16,11 +16,11 @@ internal class AddProjectsToSolutionPostAction : PostActionProcessor2Base, IPost
 
         public bool Process(IEngineEnvironmentSettings environment, IPostAction action, ICreationEffects creationEffects, ICreationResult templateCreationResult, string outputBasePath)
         {
-            if (string.IsNullOrEmpty(outputBasePath))
+            if (string.IsNullOrWhiteSpace(outputBasePath))
             {
-                Reporter.Error.WriteLine(string.Format(LocalizableStrings.AddProjToSlnPostActionUnresolvedSlnFile));
-                return false;
+                throw new ArgumentException($"'{nameof(outputBasePath)}' cannot be null or whitespace.", nameof(outputBasePath));
             }
+            outputBasePath = Path.GetFullPath(outputBasePath);
```

Comment:
Can this be extracted to base or common helper?
I'd add validation of absolute path (`IsPathRooted`) as well.

Reply:
Extracted. Also extracted the parsing of `targetFiles` like arguments. `AddReference` now has a bit different loop, but I don't think the functionality is affected. I don't think that restricting `targetFiles` like it was before makes sense.